pkg/kubectl/drain/drain.go

```diff
@@ -39,14 +44,20 @@ const (
 type Helper struct {
 	Client              kubernetes.Interface
 	Force               bool
-	DryRun              bool
 	GracePeriodSeconds  int
 	IgnoreAllDaemonSets bool
 	Timeout             time.Duration
 	DeleteLocalData     bool
 	Selector            string
 	PodSelector         string
+	Out                 io.Writer
 	ErrOut              io.Writer
+
+	// TODO(justinsb): unnecessary?
+	DryRun bool
+
+	// OnPodDeletedOrEvicted is called when a pod is evicted/deleted; for printing progress output
+	OnPodDeletedOrEvicted func(pod *corev1.Pod, usingEviction bool)
```

Comment:
Should we change this to a boolean flag to print using the klogr interface (as suggested above)?

Reply:
Personally I'd rather create an io.Writer -> klogr mapping so that we just always use klogr, but people that wanted to use an io.Writer would have an easy way of doing so.

There's an argument that klogr also allows for reformatting (and translation) of output..